ideascube/serveradmin/catalog.py

```diff
@@ -126,6 +129,10 @@ def remove(self, package):
         self._removed.append(package)
 
     def commit(self):
+        visible_packages = get_config('ideascube', 'displayed-package-ids')
+        [visible_packages.remove(p.id) for p in self._removed if p.id in visible_packages]
+        visible_packages.extend(p.id for p in self._installed)
+        set_config('ideascube', 'displayed-package-ids', visible_packages, User.objects.get_system_user())
```

Comment:
Yes we could move the dependency to the DB in the Catalog.

However, your second proposition to let Ideascube extend the Catalog to make the DB operation seems overengineering right now.
It may be a good idea when we move the catalog out of ideascube but now, it will end with a base class never directly used and extended by only one child class.